plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -212,16 +222,24 @@ export class EventsProcessor {
 
         const elementsChain = elements && elements.length ? elementsToString(elements) : ''
 
-        const personInfo = await this.db.getPersonData(teamId, distinctId)
         const groupProperties = await this.db.getGroupProperties(teamId, this.getGroupIdentifiers(properties))
 
+        let eventPersonUuid: string | null = null
         let eventPersonProperties: string | null = null
+        let personInfo = preIngestionEvent.person
+
         if (personInfo) {
-            // For consistency, we'd like events to contain the properties that they set, even if those were changed
-            // before the event is ingested. Thus we fetch the updated properties but override the values with the event's
-            // $set properties if they exist.
-            const latestPersonProperties = personInfo ? personInfo?.properties : {}
-            eventPersonProperties = JSON.stringify({ ...latestPersonProperties, ...(properties.$set || {}) })
+            eventPersonUuid = personInfo.uuid
+            eventPersonProperties = JSON.stringify(personInfo.properties)
+        } else {
+            personInfo = await this.db.getPersonData(teamId, distinctId)
+            if (personInfo) {
+                // For consistency, we'd like events to contain the properties that they set, even if those were changed
+                // before the event is ingested. Thus we fetch the updated properties but override the values with the event's
+                // $set properties if they exist.
+                const latestPersonProperties = personInfo ? personInfo?.properties : {}
+                eventPersonProperties = JSON.stringify({ ...latestPersonProperties, ...(properties.$set || {}) })
```

Comment:
cc @yakkomajuri note that this logic here is incorrect and does not handle $set_once properties. Should I file a bug here?

I think I can 'incidentally' solve this once I move processing outside of here.

Also, no unit tests for this function = :(